### ISSUE TITLE
How can I fully verify my application with call back functions?

### ISSUE THREAD

**Issue by merbin2012** [javiersantos/PiracyChecker#9]
Actually I am a cordova developer and I have tried many cordova plugins but no use, so I have decided to implement this license check process in java code, I tried many pugins but nothing was run successfully. But finally I found your plugin, really it is awsome, it was run successfully without any errors. 
And I have two questions
1) I have .jks file and how can I get the signing certification from that?
2) **enableGooglePlayLicensing** does not call the call back functions.

**How?**
```Java
new PiracyChecker(this).callback(new PiracyCheckerCallback() {
            @Override
            public void allow() {
                new AlertDialog.Builder(MainActivity.this)
                        .setTitle("License Check")
                        .setMessage("License is Valid")
                        .show();
            }

            @Override
            public void dontAllow(PiracyCheckerError error) {
                new AlertDialog.Builder(MainActivity.this)
                        .setTitle("License Check")
                        .setMessage("Not Valid License")
                        .show();
            }
        })
                .enableGooglePlayLicensing("my 64 BIT KEY")
                .start();
```

But this code does not trigger any one of the call back method.

Please tried to reply soon as possible.
Thanks Advance.

**Comment by jahirfiquitiva** [comment]
@merbin2012 read the README, it explains how to get the signing certificate

**Comment by merbin2012** [comment]
How can I get the license error details like "old license key, expired license key, and etc..."

**Comment by jahirfiquitiva** [comment]
@merbin2012 If I'm not wrong, Google only provides one KEY, so there won't be old or expired licenses.
If you want them with the signing key, if you use a expired key, I think the apk won't be generated, or you won't be able to publish it on Play Store, so it's useless too.

**Comment by merbin2012** [comment]
I have another doubt, after implementing this code I have tried to decompile this apk, unbelievable  all my java code are visible. So everyone can just remove this license validation code and they can use it life time free, is any way to protect my code? I have tried proguard but it is just rename the method name, so that is also unusable.

**Comment by javiersantos** [comment]
ProGuard should rename classes, fields and methods if it is properly configured with ProGuard rules. Keep in mind that the more obfuscated is the code, the more difficult will be to debug errors in production.
